src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1123,7 +1123,7 @@ public DecisionDag(DagState rootNode)
             private static ImmutableArray<DagState> Successor(DagState state)
             {
 
-                if (state.TrueBranch != null && state.FalseBranch != null)
+                if (state is { TrueBranch: { }, FalseBranch: { } })
```

Comment:
fwiw i don't view that is being substantively better.  i feel like it takes more mental energy to figure out what this is tryin to say over the clear and obvious expression-oriented code.

Reply:
This is a valid application of the refactoring, so the user has the option to make this decision.